draft-ietf-quic-recovery.md

```diff
@@ -762,47 +762,95 @@ The minimum congestion window is the smallest value the congestion window can
 decrease to as a response to loss, ECN-CE, or persistent congestion.
 The RECOMMENDED value is 2 * max_datagram_size.
 
-## Slow Start
-
-While in slow start, QUIC increases the congestion window by the
-number of bytes acknowledged when each acknowledgment is processed, resulting
-in exponential growth of the congestion window.
-
-QUIC exits slow start upon loss or upon increase in the ECN-CE counter.
-When slow start is exited, the congestion window halves and the slow start
-threshold is set to the new congestion window.  QUIC re-enters slow start
-any time the congestion window is less than the slow start threshold,
-which only occurs after persistent congestion is declared.
-
-## Congestion Avoidance
-
-Slow start exits to congestion avoidance.  Congestion avoidance uses an
-Additive Increase Multiplicative Decrease (AIMD) approach that increases
-the congestion window by one maximum packet size per congestion window
-acknowledged.  When a loss or ECN-CE marking is detected, NewReno halves
-the congestion window, sets the slow start threshold to the new
-congestion window, and then enters the recovery period.
-
-## Recovery Period {#recovery-period}
-
-A recovery period is entered when loss or ECN-CE marking of a packet is
-detected in congestion avoidance after the congestion window and slow start
-threshold have been decreased.  A recovery period ends when a packet sent
-during the recovery period is acknowledged.  This is slightly different from
-TCP's definition of recovery, which ends when the lost packet that started
-recovery is acknowledged.
+## Congestion Control States
+
+The congestion controller has three distinct states, as shown in
+{{fig-cc-fsm}}.
+
+~~~
+                 Start or         +------------+
+            persistent congestion |   Slow     |
+        (O)---------------------->|   Start    |
+                                  +------------+
+                                        |
+                                Loss or |
+                        ECN-CE increase |
+                                        v
+ +------------+     Loss or       +------------+
+ | Congestion |  ECN-CE increase  |  Recovery  |
+ | Avoidance  |------------------>|   Period   |
+ +------------+                   +------------+
+           ^                            |
+           |                            |
+           +----------------------------+
+              Acknowledgment of packet
+                sent during recovery
+~~~
+{: #fig-cc-fsm title="Congestion Control States and Transitions"}
+
+These states and the transitions between them are described in subsequent
+sections.
+
```

Comment:
Perhaps add a note about persistent congestion, since that's not an explicit state in this machine. Maybe something like "Note that a sender can declare persistent congestion in any of these states (see {{persistent-congestion}}), after which it enters slow start.".